src/openassetio-python/package/openassetio/managerApi/ManagerInterface.py

```diff
@@ -479,85 +479,104 @@ def finalizedEntityVersion(self, entityRefs, context, hostSession, overrideVersi
     #
     # @{
 
-    @abc.abstractmethod
-    def getRelatedReferences(
-        self, entityRefs, relationshipTraitsDatas, context, hostSession, resultTraitSet=None
+    def getWithRelationship(
+        self, relationshipTraitsData, toEntityReferences, context, hostSession, resultTraitSet=None
     ):
         """
-        Returns related entity references, based on a relationship
-        defined by a set of traits and their properties.
+        Return entity references that are related to the input
+        references by the relationship defined by a set of traits and
+        their properties.
 
         This is an essential function in this API - as it is widely used
-        to query organisational hierarchy, etc...
+        to query other entities or organisational structure.
 
-        There are three possible conventions for calling this function,
-        to allow for batch optimisations in the implementation and
-        prevent excessive query times with high-latency services.
+        @note The order of entities in the inner lists of matching
+        references should not be considered meaningful, but the outer
+        list should match the input entity reference order.
 
-         - a)  A single entity reference, a list of relationships.
-         - b)  A list of entity references and a single relationship.
-         - c)  Equal length lists of references and relationship.
+        If any relationship definition is unknown, then an empty list
+        should be returned for that entity, and no errors raised. The
+        default implementation returns an empty list for all
+        relationships.
 
-        In all cases, the return value is a list of lists, for example:
+        @param relationshipTraitsData @fqref{TraitsData} "TraitsData"
+        The traits of the relationship to query.
 
-            a)  getRelatedReferences([ r1 ], [ td1, td2, td3 ])
+        @param toEntityReferences `List[` @fqref{EntityReference} "EntityReference" `]`
```

Comment:
[Consider] I'm guessing `toEntityReferences` is gesturing at the phrase, "relationship to entity reference"? Took me a while to figure out, not entirely sure it's clear and may be adding more confusion. Would personally have this remain as `entityRefs`

Reply:
Cool - will revert :-)

Reply:
Reverted https://github.com/OpenAssetIO/OpenAssetIO/pull/915/commits/2e18cf33877d34b7ce54d266fef5b9115b3cbd00